lib/bolt/executor.rb

```diff
@@ -133,10 +133,12 @@ def queue_execute(targets)
           rescue StandardError, NotImplementedError => e
             result_promises.each do |target, promise|
               # If an exception happens while running, the result won't be logged
-              # by the CLI. Log a warning, as this is probably a problem with the transport.
+              # by the CLI. Unless the error is explicitly raised because a feature is not
+              # supported by the transport, log a warning as this is probably a problem
+              # with the transport.
               # If batch_* commands are used from the Base transport, then exceptions
               # normally shouldn't reach here.
-              @logger.warn(e)
+              @logger.warn(e) unless e.is_a?(Bolt::NotSupportedError)
```

Comment:
Hmm actually I think the right thing to do is to make the Orchestrator transport return `Bolt::Result` objects that indicate the problem, rather than throwing an exception. That matches the behavior provided by the default implementation of the `batch_*` methods in `Bolt::Transport::Base#with_events`. Then this top level handler is really just for unhandled exceptions which probably _are_ a problem with the transport, as the comment above says.

That establishes/maintains a contract that the result of calling a `batch_*` method on a transport is a list of `Bolt::Result` objects.